DDD-12.md

```diff
@@ -0,0 +1,322 @@
+
+- [1.Debezium Engine Lifecycle events](#quarkus-debezium-lifecycle-events)
+- [2.Debezium Heartbeat events](#quarkus-debezium-heartbeats-events)
+- [3.Debezium Listener](#quarkus-debezium-listener)
+- [4.Debezium Custom Data Converter](#custom-debezium-data-converter)
+- [5.Debezium SchemaChange Listener](#quarkus-debezium-schemachange-listener)
+- [6.Debezium Notification Handler](#quarkus-debezium-notification-handler)
+- [7.Debezium PostProcess Consumer](#quarkus-debezium-postprocessor)
+- [8.Debezium Custom Converter](#quarkus-debezium-custom-converter)
+
+### Quarkus Debezium Lifecycle Events
+
+We can summarize the lifecycle of a Debezium Embedded Engine in the follows steps:
+
+| Phase            | description                                                           | code                                  |
+|------------------|-----------------------------------------------------------------------|---------------------------------------|
+| *initialization* | configuration is built and the engine is created, but not yet running | after the configuration is validated  |
+| *startup*        | connectors are initialized, DB connection established                 | right after `engine.run()` is invoked |
+| *shutdown*       | engine terminates (graceful or with error)                            | observable via `CompletitionCallback` |
+
+The Quarkus Debezium extension allows you to be notified of the engine's state using the following annotation:
+
+```java
+import io.debezium.engine.source;
+import jakarta.enterprise.context.ApplicationScoped;
+
+
+@ApplicationScoped
+class DebeziumEngineLifeCycle {
+    
+    @DebeziumEngineInit()
+    public void init() {
+        /// some logic to apply
+    }
+
+    @DebeziumEngineStartup()
+    public void startup(DebeziumSourceConnectorContext context) {
+        /// some logic to apply 
+    }
+
+    @DebeziumEngineShutdown()
+    public void shutdown(Status status) {
+        /// some logic to apply
+    }
+}
+```
+
+### Quarkus Debezium Heartbeats events
+
+In Debezium, heartbeat events are lightweight, periodic messages emitted when no database changes occur. They confirm that the connector is alive and still connected to the source, helping to detect liveness. In the Quarkus Debezium Extension can detect heartbeats events in this way:
+
+```java
+import io.debezium.engine.ChangeEvent;
+import jakarta.enterprise.context.ApplicationScoped;  
+
+
+@ApplicationScoped  
+class HeartbeatListener {
+  
+    @DebeziumHeartbeat()  
+    public void heartbeat(ChangeEvent<String, String> event) {  
+        /// some logic to apply 
+    }  
+}
+```
+
+it's possible to define the `heartbeat.action.query` and `heartbeat.interval.ms` interval in case of relational databases:
+
+```java
+import io.debezium.engine.ChangeEvent;
+import jakarta.enterprise.context.ApplicationScoped;  
+
+
+@ApplicationScoped  
+class HeartbeatListener {
+  
+    @DebeziumHeartbeat(query="SELECT now()", interval=10000)  
```

Comment:
Let's say I want to create a reusable heartbeat handler that sends events to a Kafka topic or another target type, then I should be able to define that class with a method that `@Observes DebeziumHeartbeat`. The point of this class is not to define configuration, but to define the handler topic irrespective of how the heartbeat is triggered. This allows maximum reuse. If we mix the configuration concern here, I find that's breaking SoC.  wdyt?